### PR TITLE
[BugFix] fix missed cpu cores states on follower FE

### LINKED ISSUE #5789: fix missed cpu cores states on follower FE
### Problem
Current pipeline dop calculation and statistics relies on the knowledge of CPU cores of backends, which should be reported by backend. But which has several drawbacks:
1. No interface on FE could be used to examine the cpu cores of backends
2. Only master FE could retrieve this information, follower FEs has no knowledge about it

This bug would lead to:
1. At follower FE, the pipeline execution engine could work correctly, due to the lack of cpu core states

### Fix
Severals steps should be adopt to fix it:
1. Display `cpuCores` of backends in `show backends` command
3. Transmit the cpuCores to followers and observers through the edit-log
  1. Change the serialization format to JSON, to make it forward-compatible
  2. Add a new message `OP_HEARTBEAT_V2`, to make it backward-compatible
  4. Add a field `cpuCores` to `BackendHbResponse`

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5789

## Problem Summary(Required) ：

Fix the missed `cpuCores` states on follower FE.

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatResponse.java [hunk 1]
```diff
@@ -98,8 +102,7 @@ public static HeartbeatResponse read(DataInput in) throws IOException {
 
     @Override
     public void write(DataOutput out) throws IOException {
-        Text.writeString(out, type.name());
-        Text.writeString(out, status.name());
+        Text.writeString(out, GsonUtils.GSON.toJson(this));
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatResponse.java:105]: why write with json bug read without json?

File: fe/fe-core/src/main/java/com/starrocks/system/BackendCoreStat.java [hunk 2]
```diff
@@ -6,6 +6,8 @@
 import java.util.concurrent.atomic.AtomicInteger;
 
 public class BackendCoreStat {
+    private static final int DEFAULT_CORES_OF_BE = 1;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/system/BackendCoreStat.java:9]: 8 or 16?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 16 / 28 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatResponse.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/persist/HbPackage.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendCoreStat.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendHbResponse.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/persist/HbPackage.java [hunk 3]
```diff
@@ -52,14 +55,17 @@ public static HbPackage read(DataInput in) throws IOException {
...
     public void write(DataOutput out) throws IOException {
-        out.writeInt(hbResults.size());
-        for (HeartbeatResponse heartbeatResult : hbResults) {
-            heartbeatResult.write(out);
-        }
+        Text.writeString(out, GsonUtils.GSON.toJson(this));
     }
 
+    // Deprecated
     public void readFields(DataInput in) throws IOException {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/persist/HbPackage.java:69]: use annotation

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatResponse.java (hunk 1 above)
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatResponse.java:105]: You changed the format. The new version process won't be compatible with old format. Is this OK?

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 16 / 28 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatResponse.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/persist/HbPackage.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendCoreStat.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendHbResponse.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|1|100.00%|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 16 / 28 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatResponse.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/persist/HbPackage.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendCoreStat.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendHbResponse.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 16 / 28 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatResponse.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/persist/HbPackage.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendCoreStat.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/system/BackendHbResponse.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|1|100.00%|

**Pull request merged**